torch/utils/_pytree.py

```diff
@@ -323,24 +335,22 @@ def register_constant(cls: type[Any]) -> None:
 
     """
 
-    def _flatten(x):  # type: ignore[no-untyped-def]
-        return [], ConstantNode(x)
-
-    def _unflatten(_, context):  # type: ignore[no-untyped-def]
-        return context.value
-
-    def _flatten_with_keys(x):  # type: ignore[no-untyped-def]
-        return [], ConstantNode(x)
-
     _private_register_pytree_node(
         cls,
-        _flatten,
-        _unflatten,
-        flatten_with_keys_fn=_flatten_with_keys,
+        _constant_flatten,
+        _constant_unflatten,
+        flatten_with_keys_fn=_constant_flatten_with_keys,
     )
 
 
-@dataclasses.dataclass
+def is_constant_cls(cls: type[Any]) -> bool:
+    node = SUPPORTED_NODES.get(cls, None)
```

Comment:
Can't we use a new global set `CONSTANT_NODES`?

Reply:
I have no preference, this patch is just moving existing code a bit and I wanted to avoid breaking things.

Reply:
I mean we can have a new global variable to track for this.

```python
CONSTANT_NODES: set[type] = set()


def register_constant(cls):
    _private_register_pytree_node(cls, ...)
    CONSTANT_NODES.add(cls)


def is_constant_class(cls):
    return isinstance(cls, type) and cls in CONSTANT_NODES
```

Reply:
I see, sure I can add that.